commands/command_checkout.go

```diff
@@ -14,6 +15,15 @@ import (
 
 func checkoutCommand(cmd *cobra.Command, args []string) {
 	requireInRepo()
+
+	msg := []string{
+		"WARNING: 'git lfs checkout' is deprecated and will be removed in v3.0.0.",
+
+		"'git checkout' has been updated in upstream Git to have comparable speeds",
```

Comment:
Minor nit: We could say here `upstream Git 2.15+`. If you like that then I would make a PR with an according commit. If you think that's not necessary then this is great too.

The PR is 👍 👍

Reply:
If you feel strongly, I certainly encourage it, but I think that this is fine without it. `git lfs clone` only shows this message based on the Git version, which I think we should not replicate here.